esmvaltool/diag_scripts/ocean/diagnostic_maps_quad.py

```diff
@@ -0,0 +1,272 @@
+    extract_levels:
+      levels:  [100., ]
+      scheme: linear_extrap
+    time_average:
+
+This tool is part of the ocean diagnostic tools package in the ESMValTool,
+and was based on the plots produced by the Ocean Assess/Marine Assess toolkit.
+
+Author: Lee de Mora (PML)
+        ledm@pml.ac.uk
+        """
+import logging
+import os
+import sys
+import matplotlib
+matplotlib.use('Agg')  # noqa
+import iris
+import cartopy
+
+import matplotlib.pyplot as plt
+import iris.quickplot as qplt
+import numpy as np
+
+import diagnostic_tools as diagtools
+from esmvaltool.diag_scripts.shared import run_diagnostic
+
+# This part sends debug statements to stdout
+logger = logging.getLogger(os.path.basename(__file__))
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+
+def makemapplot(fig, ax, lons, lats, data, title, zrange=[-100, 100],
+                lon0=0., drawCbar=True, cbarlabel='', doLog=False, ):
+
+    if len(lons) == 0:
+        return fig, ax
+    try:
+        if len(lons.compressed()) == 0:
+            return False, False
+    except AttributeError:
+        logger.warning('makemapplot: latitude and longitude not provided.')
+
+    lons = np.array(lons)
+    lats = np.array(lats)
+    data = np.ma.array(data)
+
+    if doLog and zrange[0] * zrange[1] <= 0.:
+        data = np.ma.masked_less_equal(ma.array(data), 0.)
+
+    if data.ndim == 1:
+        if doLog:
+            im = ax.scatter(lons, lats, c=data, lw=0, marker='s',
+                            transform=cartopy.crs.PlateCarree(),
+                            norm=LogNorm(),
+                            vmin=zrange[0],
+                            vmax=zrange[1])
+        else:
+            im = ax.scatter(lons, lats, c=data, lw=0, marker='s',
+                            transform=cartopy.crs.PlateCarree(),
+                            vmin=zrange[0],
+                            vmax=zrange[1])
+    else:
+        crojp2, data, newLon, newLat = regrid(data, lats, lons)
+
+        if doLog:
+            im = ax.pcolormesh(newLon, newLat, data,
+                               transform=cartopy.crs.PlateCarree(),
+                               norm=LogNorm(vmin=zrange[0],
+                                            vmax=zrange[1]), )
+        else:
+            im = ax.pcolormesh(newLon, newLat, data,
+                               transform=cartopy.crs.PlateCarree(),
+                               vmin=zrange[0], vmax=zrange[1])
+
+    ax.add_feature(cartopy.feature.LAND,  facecolor='0.85')
+
+    if drawCbar:
+        c1 = fig.colorbar(im, pad=0.05, shrink=0.75)
+        if len(cbarlabel) > 0:
+            c1.set_label(cbarlabel)
+
+    plt.title(title)
+
+    ax.set_axis_off()
+    plt.axis('off')
+    ax.axis('off')
+
+    return fig, ax
+
+
+def match_moddel_to_key(m, cfg_dict, input_files_dict, ):
+    """Match up the three models and observations dataset from the configs."""
+    if not isinstance(cfg_dict, dict):
+        print('problem!', m)
```

Comment:
try not to use ```print``` but rather a logger message, also make it more explicit what went wrong